crypto/threads_win.c

```diff
@@ -207,14 +207,42 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
 int CRYPTO_atomic_or(uint64_t *val, uint64_t op, uint64_t *ret,
                      CRYPTO_RWLOCK *lock)
 {
-    *ret = (uint64_t)InterlockedOr64((LONG64 volatile *)val, (LONG64)op) | op;
-    return 1;
+#if (defined(_MSC_VER) && defined(_M_IX86) && _MSC_VER <= 1500)
+    // VC++ 2008 or earlier x86 compilers do not have an inline implementation of InterlockedOr64 for 32bit and will fail to run on Windows XP 32bit.
+    // See: https://docs.microsoft.com/en-us/cpp/intrinsics/interlockedor-intrinsic-functions#requirements
+    // To work around this problem, we implement a manual locking mechanism for only VC++ 2008 or earlier x86 compilers.
+	if (lock == NULL || !CRYPTO_THREAD_write_lock(lock))
```

Comment:
Please use C98 block comments, not C99 line comments, and wrap lines to 80 characters.

Reply:
What about doing this at the top of the file once...

````
/*
 * Comments.... 
 */
#if (defined(_MSC_VER) && defined(_M_IX86) && _MSC_VER <= 1500)
# define NO_INTERLOCKEDOR64
#endif
````

Reply:
This patch is also required for VS 2015 which is _MSC_VER = 1900

I just changed the pragma check for _MSC_VER <= 1900 in crypto\thread_win.c and the build completed using VS2015.